classes/class-ledyer-lco-gateway.php

```diff
@@ -339,6 +350,56 @@ public function process_payment_handler( $order_id ) {
 			return false;
 		}
 
+		/**
+		 * Process the payment for HPP/redirect checkout flow.
+		 *
+		 * @param object $order The WooCommerce order.
+		 *
+		 * @return array|string[]
+		 */
+		protected function hpp_redirect_handler( $order ) {
+
+			if ( empty( $order ) ) {
+
+				return array(
+					'result'   => 'error',
+					'messages' => array( 'Failed to get order for HPP.' ),
+				);
+			}
+			$this->process_payment_handler( $order->get_id() );
```

Comment:
Should this function be called even for the HPP flow? Since right now it tries to get the Ledyer session, but thats created later.